tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,370 @@
+Write-Host -ForegroundColor Magenta "Generated release log from tag $lastTag to the current HEAD."
+Write-Host -ForegroundColor White "The detailed log can be found by editing:" (Get-ChildItem $releaseLogDetailed).FullName
+Write-Host -ForegroundColor White "The short log can be found by editing:"  (Get-ChildItem $releaseLogShort).FullName
+Write-Host
+Write-Host -ForegroundColor Cyan "NOTE: If there is a tag that you want to compare to that is earlier than the most recent tag you can run:"
+Write-Host -ForegroundColor Yellow "git log --stat <tagversion>..HEAD --output releaselog_detailed.txt"
+Write-Host -ForegroundColor Yellow "git log --oneline <tagversion>..HEAD --output releaselog_short.txt"
+Write-Host
+
+# Create a list of the markdown files so we can compare them to the API doc directory
+for ($assemblyIndex = 0; $assemblyIndex -lt $assemblyRootNames.length; $assemblyIndex++) 
+{ 
+    if (Test-Path $assemblyFilePath[$assemblyIndex]) 
+    {
+        $assemblyFileToUse = $assemblyFilePath[$assemblyIndex]
+        $markdownOutputFileToUse = $markdownOutputFilePath[$assemblyIndex]
+        
+        # Get the original header from the file so we can apply it to the newly generated file.
+        # Grabs the first 5 lines of the file which generally looks like this...
+        # Azure SDK .NET Public API
+        #
+        # ## Microsoft.Azure.Devices.Client 1.35.*
+        #
+        # ```C
+        $originalMarkdownHeader = Get-Content $markdownOutputFilePath[$assemblyIndex] | Select-Object -First 5
+        Write-Debug "Original markdown header to replace in new file"
+
+        foreach ($outLine in $originalMarkdownHeader) 
+        {
+            Write-Debug $outLine
+        }
+        
+        # Permalink for AsmDiff README is: https://github.com/dotnet/arcade/blob/3aea914072c2f8844d7cf74c41c759b497e59b16/src/Microsoft.DotNet.AsmDiff/README.md
+        #
+        # These asmToolSwitches do the following
+        # -os <filename>    Specifies the dll we want to generate the markdown for
+        # -w Markdown       Tells the tool to generate Markdown output
+        # -o <filename>     Specifies the name of the markdown file to output
+        # -gba              Flattens the name spaces and removes the namespace headers from the output (ex. ## Microsoft.Azure.Devices.Client)
+        Write-Host -ForegroundColor Magenta "Creating markdown for $assemblyFileToUse"
+        $asmToolSwitches = "-os", $assemblyFileToUse, "-w", "Markdown", "-o", $markdownOutputFileToUse, "-gba"
+        & $asmToolExecutableCommand $asmToolSwitches
+        
+        # Replace the header for this file using the original header
+        $newMarkdownBodyContent = Get-Content $markdownOutputFileToUse | Select-Object -Skip 5
+        . {
+            $originalMarkdownHeader
+            $newMarkdownBodyContent
+        } | Set-Content $markdownOutputFileToUse
+    } 
+    else 
+    {
+        Write-Host $assemblyFileToUse "does not exist. Skipping."
```

Comment:
consider addressing the error case first, so the reason we hit this condition isn't so many lines away from the code that handles it.

You could then use the `continue;` keyword, and not need the happy path inside an if/else statement.